ADR/ADR-001-Git-Branching-Model.md

```diff
@@ -0,0 +1,225 @@
+* Deploy (possibly automatically) or Publish artifact
+
+**Advantages**
+
+* Simplicity: Fewer branches to manage and understand
+* Faster delivery: Direct path from feature to production
+* Reduced merge conflicts: Shorter-lived branches mean fewer conflicts
+* Continuous integration: Encourages frequent integration with main
+* Lower cognitive overhead: Developers focus on main + feature branch
+* Better for small teams: Less coordination overhead
+* Encourages quality: Every merge must be production-ready (although this can also increase risk)
+* Ideal for continuous delivery: Aligns with systems that deploy frequently
+* Pull request-centric: Built-in code review and collaboration
+
+**Disadvantages**
+
+* Higher risk: Main branch may become unstable during active development
+* Quality gates dependency: Relies very heavily on a complete CI/CD, requiring robust automated testing to ensure main is always deployable 
+* Limited release preparation: No dedicated space for release stabilization
+* Coordination challenges: Requires strict discipline for production readiness
+* Hotfix complexity: May interfere with ongoing feature development
+* Less suitable for scheduled releases: Harder to prepare release candidates
+* Not ideal for versioned software: Difficult to maintain multiple versions
+* Less structured: No clear separation between production and development code
+* Requires strong team discipline: Teams must ensure main is always deployable
+* Not as common in operational software: Many operational software packages use Git Flow
+
+### Analysis
+
+**Team Size Impact**
+
+* Small teams (1-3 developers): GitHub Flow reduces overhead and simplifies collaboration
+* Medium teams (4-8 developers): Either approach viable, depends on release cadence and software type
+* Large teams (9+ developers): Git Flow provides better coordination and parallel development structure
+
+**Release Patterns**
+
+* Continuous deployment: GitHub Flow aligns perfectly with continuous delivery practices
+* Scheduled releases: Git Flow provides better control and release preparation capabilities
+* Mixed patterns: May require different strategies per repository based on deployment needs
+
+**Software Type Considerations**
+
+* Web applications: GitHub Flow ideal for continuously delivered web apps (when appropriate CI/CD is in place)
+* Versioned software: Git Flow better suited for software requiring multiple version support (especially if the software is operational)
+* Simple applications: GitHub Flow avoids unnecessary complexity
+
+**Operational Complexity**
+
+* Simple applications: GitHub Flow reduces unnecessary complexity
+* Critical systems: Git Flow provides additional safety measures and structured release process
+* Evolving requirements: Start simple, add complexity as needed
+
+## Decision
+
+We will adopt a **hybrid approach** that allows repositories to evolve their branching strategy based on operational needs and complexity:
```

Comment:
Do we have a pathway to do the reverse migration - from git-flow to GitHub-flow - where the additional complexity does not bring us value.

I am thinking particularly of:

- Bundle repos. Essentially the only commits in these repos are changing version numbers and then retagging.
- (Personally) ECFS where both for the client and the server, every single release is just merging develop into master and tagging...